ArduCopter/mode.h

```diff
@@ -1324,7 +1326,8 @@ class ModeLoiter : public Mode {
     bool init(bool ignore_checks) override;
     void run() override;
 
-    bool requires_GPS() const override { return true; }
+    bool requires_GPS() const override { return false; }
```

Comment:
Thanks for this, it would definitely be nice if users didn't need to turn off arming checks when using optical flow (or other non-GPS based position estimation).

I think "requires_GPS" and "requires_position()" should always be the same.  The "requires_**GPS**" is just a hang-over from when GPS was the only viable option.  If we agree on this then let's just rename, "requires_GPS" to "requires_position".

Reply:
Andy and I have discussed this elsewhere.

I was pushing for `requires_location` and `requires_position`, using the same terms we use elsewhere - `Location` as in absolute lat/lon object and "position" as in offsets-from-origin.

`requires_absolute_location` and `requires_position_relative_to_origin` could also work :-)

... or `requires_latitude_longitude_altitude`?